aocd/get.py

```diff
@@ -73,7 +79,7 @@ def current_day():
     return day
 
 
-def get_day_and_year():
+def get_day_and_year() -> tuple[int, Optional[int]]:
```

Comment:
The year should not be optional. I think we probably really want `tuple[int, int]` here, with whatever refactor is necessary to make that happen. Either `get_day_and_year` should raise if it can not introspect both, or it should use `most_recent_year()` if it can not otherwise determine a year, whichever is more backwards compat with existing behaviour.